core/timeout_handler.cpp

```diff
@@ -35,8 +35,17 @@ void TimeoutHandler::refresh(const void *cookie)
 
     auto it = _timeouts.find((void *)(cookie));
     if (it != _timeouts.end()) {
-        dl_time_t future_time = steady_time_in_future(it->second->duration_s);
-        it->second->time = future_time;
+        it->second->start_time = steady_time();
+    }
+}
+
+void TimeoutHandler::update(double new_duration_s, const void *cookie)
```

Comment:
what about `updated_duration_s` instead of `new_duration_s`?

Reply:
Sure, thanks. See ea4cdb8.